components/Common/Dropdown/index.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useRef } from 'react';
+import { useElementPositionAndSize } from '../../../hooks/useElementPositionAndSize';
+import { computeDropdownStyles } from './computeDropdownStyles';
+import { DropdownItem } from '../../../types';
+import styles from './index.module.scss';
+
+interface Props<T> {
+  items: Array<DropdownItem>;
+  shouldShow: boolean;
+  elementRef: React.RefObject<T>;
+}
+
+const Dropdown = <T extends HTMLElement>({
+  items,
+  shouldShow,
+  elementRef,
+}: Props<T>): JSX.Element | null => {
```

Comment:
You don't need `JSX.Element` here. You can remove this explicit return type from here :)